packages/iov-keycontrol/src/proposal.ts

```diff
@@ -0,0 +1,176 @@
+/*
+Code for https://github.com/iov-one/iov-core/issues/302
+
+This file just contains a proposal for a new API for this package.
+This code will not be run, but by placing it here, we force
+it to be tslint/tsc compatible.
+
+Once approved, this should be a template to update the other files in
+this package.
+*/
+
+import {
+  Nonce,
+  PrehashType,
+  SignableBytes,
+  SignedTransaction,
+  TxCodec,
+  UnsignedTransaction,
+} from "@iov/bcp-types";
+import { Slip0010RawIndex } from "@iov/crypto";
+import { Algorithm, ChainId, PrivateKeyBytes, PublicKeyBytes, SignatureBytes } from "@iov/tendermint-types";
+
+import { KeyringEntryImplementationIdString, PublicIdentity } from "./keyring";
+import { ValueAndUpdates } from "./valueandupdates";
+
+export interface LocalIdentity extends PublicIdentity {
+  readonly label?: string;
+}
+
+export interface LocalHDIdentity extends LocalIdentity {
+  readonly path: ReadonlyArray<Slip0010RawIndex>;
+}
+
+// some utilities like clone, serialize removed for simplicity just
+// to focus on key management api.
+// this is good enough for signing, for creating keys, you need
+// a refinement (HDKeyringEntry or MixedKeyringEntry)
+export interface KeyringEntry<T extends PublicIdentity = LocalIdentity> {
+  // these store labels locally outside of the key material
+  // do we want Promises for the case of proxy connections?
+  readonly setLabel: (label: string | undefined) => void;
```

Comment:
seems like  this should be `setKeyringLabel` to more clearly and in parallel separate for the average user from `setIdentityLabel` because these two concepts are very easy to confuse and not so clear for anybody yet.